ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/TransactionCommitter.scala

```diff
@@ -376,28 +376,37 @@ private[kvutils] class TransactionCommitter(
   private def buildFinalLogEntry: Step =
     (commitContext, transactionEntry) => StepStop(buildLogEntry(transactionEntry, commitContext))
 
-  private def validateContractKeys: Step = (commitContext, transactionEntry) => {
+  private[committer] def validateContractKeys: Step = (commitContext, transactionEntry) => {
     val damlState = commitContext.collectInputs {
       case (key, Some(value)) if key.hasContractKey => key -> value
     } ++ commitContext.getOutputs
     val startingKeys = damlState.collect {
       case (k, v) if k.hasContractKey && v.getContractKeyState.getContractId.nonEmpty => k
     }.toSet
-    validateContractKeyUniqueness(commitContext.recordTime, transactionEntry, startingKeys) match {
-      case StepContinue(transactionEntry) =>
-        validateContractKeyCausalMonotonicity(
-          commitContext.recordTime,
-          transactionEntry,
-          startingKeys,
-          damlState)
-      case err => err
+    val resolvedIds: Map[DamlContractKey, String] = damlState.collect {
```

Comment:
Given that there may be several resolution pairs for the same key in the list and the order of them is unspecified, this conversion to a `Map` will not consistently pick the key resolution that must hold at the beginning of the transaction. So this will not work.

Reply:
That's unfortunately true 😞 
The same applies for the line 796:
```
    val contractKeyToIdMap: Map[DamlContractKey, String] =
      submission.getContractKeyIdPairsList.asScala.map { pair =>
        pair.getKey -> pair.getResolvedToId
      }.toMap
```

Reply:
This is irrelevant after recent changes.